astropy/table/sorted_array.py

```diff
@@ -171,27 +171,30 @@ def find(self, key: tuple) -> Sequence[Integral]:
 
     def range(
         self,
-        lower: tuple[Hashable, ...],
-        upper: tuple[Hashable, ...],
+        lower: tuple[Hashable, ...] | None,
+        upper: tuple[Hashable, ...] | None,
         bounds: tuple[bool, bool],
     ) -> list[int]:
         """
         Find values in the given range.
 
         Parameters
         ----------
-        lower : tuple
-            Lower search bound
-        upper : tuple
-            Upper search bound
+        lower : tuple, None
+            Lower search bound (no lower bound if None)
+        upper : tuple, None
+            Upper search bound (no upper bound if None)
         bounds : (2,) tuple of bool
             Indicates whether the search should be inclusive or
             exclusive with respect to the endpoints. The first
             argument corresponds to an inclusive lower bound,
             and the second argument to an inclusive upper bound.
         """
-        lower_pos = self.find_pos(lower, 0)
-        upper_pos = self.find_pos(upper, 0)
+        # Find initial positions for lower and upper bounds. Just like a slice object,
+        # None values for `lower` or `upper` correspond to no bound in that direction.
```

Comment:
here too, I don't think this is intrinsic to `slice` objects, but is a meaning we're accustomed to because of numpy semantics
https://docs.python.org/3/library/functions.html#slice